docs/docs/custom_preconditioners.md

```diff
@@ -0,0 +1,142 @@
+
+# Defining custom gradient preconditioners
+
+NetKet calls _gradient preconditioner_ that class of techniques that transform the
+gradient of the cost function in such a way to improve convergence properties before
+passing it to an optimiser like :ref:`Sgd` or :ref:`Adam`.
+
+Examples of _gradient preconditioners_ are the [Stochastic Reconfiguration](https://www.attaccalite.com/PhDThesis/html/node15.html) method  (also known as [Natural Gradient](https://wiseodd.github.io/techblog/2018/03/14/natural-gradient/), the Linear Method or second order Hessian-based optimisation.
+
+We call those methods _gradient preconditioners_ because they take as input the gradient of the cost function (usually the energy) and some additional information, and output a transformed gradient.
```

Comment:
```suggestion
We call those methods _gradient preconditioners_ because they take as input the gradient of the cost function (e.g., the energy gradient in VMC ground state optimisation) and output a transformed gradient.
```
You can leave in the "and additional information" if you want, but it's unclear in this paragraph what that means and is not important for the conceptual overview anyways.

Reply:
indeed is more clear without it